applicationset/services/scm_provider/gitea.go

```diff
@@ -0,0 +1,136 @@
+package scm_provider
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/http/cookiejar"
+	"os"
+
+	"code.gitea.io/sdk/gitea"
+)
+
+type GiteaProvider struct {
+	client      *gitea.Client
+	owner       string
+	allBranches bool
+}
+
+var _ SCMProviderService = &GiteaProvider{}
+
+func NewGiteaProvider(ctx context.Context, owner, token, url string, allBranches, insecure bool) (*GiteaProvider, error) {
+	if token == "" {
+		token = os.Getenv("GITEA_TOKEN")
```

Comment:
We have this set in case of GitHub and Gitlab, how to tackle for this for test? should we set in workflows/cibuild.yaml and before that need to setup token into secret

Reply:
We can set it up as a good citizen, but rate limiting has yet to implemented; https://github.com/go-gitea/gitea/issues/9559 . My feeling is that if we hit major issues we may want to set up a local instance in the test harness. I believe either should be handled outside of this PR.

Reply:
ok, if rate limiting is not there in gate its fine